base/docs/helpdb/Base.jl

```diff
@@ -7376,7 +7376,7 @@ for the effects of compilation.
     Array(dims)
 
 `Array{T}(dims)` constructs an uninitialized dense array with element type `T`. `dims` may
-be a tuple or a series of integer arguments. The syntax `Array(T, dims)` is also available,
+be a tuple or a series of integer arguments. The syntax `Array{T}(dims)` is also available,
```

Comment:
(The first error message says you are calling `@deprecate` with the wrong number of arguments), the second error message says you haven't fix the inference.jl for the deprecated constructor yet.